controllers/spacerequest/spacerequest_controller.go

```diff
@@ -43,12 +63,440 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, memberClusters map[strin
+	logger.Info("subSpace already exists")
+	// check if subSpace is being deleted
+	if util.IsBeingDeleted(subSpace) {
+		return false, fmt.Errorf("cannot update subSpace because it is currently being deleted")
+	}
+	// update tier and clusterroles in the subspace
+	// if they do not match the ones on the spaceRequest
+	return r.updateSubSpace(logger, subSpace, spaceRequest)
+}
+
+// validateNSTemplateTier checks if the provided tierName in the spaceRequest exists and is valid
+func (r *Reconciler) validateNSTemplateTier(tierName string) error {
+	if tierName == "" {
+		return fmt.Errorf("tierName cannot be blank")
+	}
+	// check if requested tier exists
+	tier := &toolchainv1alpha1.NSTemplateTier{}
+	if err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: r.Namespace,
+		Name:      tierName,
+	}, tier); err != nil {
+		if errors.IsNotFound(err) {
+			return err
+		}
+		// Error reading the object - requeue the request.
+		return errs.Wrap(err, "unable to get the current NSTemplateTier")
+	}
+	return nil
+}
+
+// updateSubSpace updates the tierName and targetClusterRoles from the spaceRequest to the subSpace object
+// if they are not up to date.
+// returns false/nil if everything is up to date
+// returns true/nil if subSpace was updated
+// returns false/err if something went wrong
+func (r *Reconciler) updateSubSpace(logger logr.Logger, subSpace *toolchainv1alpha1.Space, spaceRequest *toolchainv1alpha1.SpaceRequest) (bool, error) {
+	logger.Info("update subSpace")
+	if spaceRequest.Spec.TierName == subSpace.Spec.TierName &&
+		reflect.DeepEqual(spaceRequest.Spec.TargetClusterRoles, subSpace.Spec.TargetClusterRoles) {
+		// everything is up to date let's return
+		return false, nil
+	}
+
+	// update tier name and target cluster roles
+	logger.Info("updating subSpace", "name", subSpace.Name)
+	subSpace.Spec.TierName = spaceRequest.Spec.TierName
+	subSpace.Spec.TargetClusterRoles = spaceRequest.Spec.TargetClusterRoles
+	err := r.Client.Update(context.TODO(), subSpace)
+	if err != nil {
+		return false, errs.Wrap(err, "unable to update tiername and targetclusterroles")
+	}
+
+	logger.Info("subSpace updated", "name", subSpace.Name, "target_cluster_roles", subSpace.Spec.TargetClusterRoles, "tierName", subSpace.Spec.TierName)
+	return true, nil
+}
+
+// getParentSpace retrieves the name of the space that provisioned the namespace in which the spacerequest was issued.
+// we need to find the parentSpace so we can `link` the 2 spaces and inherit the spacebindings from the parent.
+func (r *Reconciler) getParentSpace(memberCluster cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest) (*toolchainv1alpha1.Space, error) {
+	parentSpace := &toolchainv1alpha1.Space{}
+	namespace := &v1.Namespace{}
+	err := memberCluster.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: "",
+		Name:      spaceRequest.Namespace,
+	}, namespace)
+	if err != nil {
+		return parentSpace, errs.Wrap(err, "unable to get namespace")
```

Comment:
let's return nil when there is an error - here and also in the other case below
```suggestion
		return nil, errs.Wrap(err, "unable to get namespace")
```